eclass/cargo.eclass

```diff
@@ -346,6 +416,39 @@ cargo_target_dir() {
 	echo "${CARGO_TARGET_DIR:-target}/$(rust_abi)/$(usex debug debug release)"
 }
 
+# @FUNCTION: cargo_update_crates
+# @USAGE:
+# @DESCRIPTION:
+# Helper function to call `cargo update --offline` with the given Cargo.toml.
+# This will update Cargo.{toml,lock}. This should provide a straightforward
+# approach to updating vulnerable crates in a package.
+#
+# To use: replace any vulnerable crates in ${CRATES} with updated (and compatible)
+# versions, then call `cargo_update_crates` in src_prepare. If Cargo.toml is not
```

Comment:
I'm confused. Is there a reason why this needs to be called explicitly, when `CRATE_PATHS_OVERRIDE` is pre-inherit?

Reply:
`CRATE_PATHS_OVERRIDE` is one way of replacing crates. If you do this you put the new crates in `CRATE_PATHS_OVERRIDE` and remove them from `CRATES`. No other action is required as the `Cargo.toml` in the crate we override with will apply to any other crate with the same name regardless of version. We need the additional `CRATE_PATHS_OVERRIDE` variable to tell the eclass which crates need their path added to ${ECARGO_HOME}/.cargo/config.toml`.

However the other method of updating a crate used by an ebuild is to change its version in `CRATES`to a non-vulnerable one, then call `cargo_update_crates` to have the package manager update the lockfile. I couldn't think of a good way of automating this, and there's no need for a separate variable for what's a one-liner either way. Strictly speaking users could implement this themselves in src_prepare; the function is just a helper: "Please update the lockfile against the newer version(s) of crate(s) available in the offline repository masquerading as crates.io"

If you have a suggestion to make this more straightforward I'm happy to hear it!

Reply:
Oh, then I got confused that you need to do both.

Reply:
I'll make sure that the 'writing rust ebuilds' wiki page gets an update that clearly explains when to use each of these (and when you'll, unfortunately, still need to patch!)